src/compile/selection/index.ts

```diff
@@ -69,12 +69,9 @@ export function forEachSelection(
 ) {
   const selections = model.component.selection;
   if (selections) {
-    for (const name in selections) {
-      if (hasOwnProperty(selections, name)) {
-        const sel = selections[name];
-        const success = cb(sel, compilers[sel.type]);
-        if (success === true) break;
-      }
+    for (const sel of vals(selections)) {
```

Comment:
Honestly, I don't like this since it requires generating an intermediate array to loop over. 

Why is `hasOwnProperty` a bad thing?

Reply:
The main thing here is that we actually only need the values and we can get much better types if we use vals. 

<img width="526" alt="Screen Shot 2020-02-23 at 13 32 59" src="https://user-images.githubusercontent.com/589034/75120545-670d2880-5641-11ea-8485-2c668d7d7b1c.png">

Reply:
Performance-wise, there isn't much difference here since the object doesn't have many keys.